uaclient/log.py

```diff
@@ -58,3 +62,35 @@ def format(self, record: logging.LogRecord) -> str:
 
         local_log_record["extra"] = extra_message_dict
         return json.dumps(list(local_log_record.values()))
+
+
+def get_user_log_file() -> str:
+    """Gets the correct user log_file storage location
+
+    returns XDG_CACHE_HOME env variable, if it is set,
+    If not set, we return the default path
+    """
+    user, _ = system.subp(["whoami"])
+    user = user.strip()
+    user_cache = os.environ.get("XDG_CACHE_HOME")
+    log_file = "{}-{}.log".format(defaults.USER_LOG_FILE_PATH, user)
+    if user_cache:
+        return (pathlib.Path(user_cache) / log_file).as_posix()
+    return (pathlib.Path.home() / ".cache" / log_file).as_posix()
+
+
+def get_users_log_file() -> List[str]:
+    """Gets all the log files for the users in the system
+
+    Returns a list of all user log files in their home directories.
+    """
+    out, _ = system.subp(["ls", "/home/"])
+    user_directories = out.splitlines()[:USER_LOG_COLLECTED_LIMIT]
```

Comment:
we can use `os.listdir()` here

aside: generally speaking, we shouldn't have to use subprocesses for almost anything. subp is a last resort for if there is no proper API to what we're trying to do. This is just a general rule and there are nuances/exceptions.